jax/experimental/jax2tf/jax2tf.py

```diff
@@ -427,6 +426,12 @@ def run_fun_flat_as_tf(
         _has_registered_tf_source_path = True
 
       if with_gradient:
+        if exported:
+          exported_vjp = exported.vjp()
+          if exported_vjp is None:
+            raise ValueError("Requested with_gradient but there is no VJP available")
```

Comment:
If we always request vjp() return Exported object,  in the definition, should we use "Optional["Exported"]]" ? Can we use "Exported" directly and move the exception raise there ?  When None return of `exported.vjp()` is used here?

Reply:
I think I understand your comment, we should move the error upstream. I have changed the handling of the "vjp()" to always return an Exported, or an exception. I think that this is in the spirit of your comment.

In truth, I do not yet understand how this will be working eventually. There is a string of several more refactorings in this area that I want to do incrementally to make it less likely that there is a regression.